queue/src/test/java/org/killbill/queue/TestReaper.java

```diff
@@ -0,0 +1,254 @@
+ * under the License.
+ */
+
+package org.killbill.queue;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.joda.time.DateTime;
+import org.killbill.CreatorName;
+import org.killbill.TestSetup;
+import org.killbill.bus.api.PersistentBusConfig;
+import org.killbill.bus.dao.BusEventModelDao;
+import org.killbill.bus.dao.PersistentBusSqlDao;
+import org.killbill.queue.api.PersistentQueueEntryLifecycleState;
+import org.skife.config.TimeSpan;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestReaper extends TestSetup {
+
+    private static final Long SEARCH_KEY_1 = 1L;
+    private static final Long SEARCH_KEY_2 = 2L;
+
+    private DBBackedQueue<BusEventModelDao> queue;
+    private PersistentBusSqlDao sqlDao;
+
+    @BeforeClass(groups = "slow")
+    public void beforeClass() throws Exception {
+        super.beforeClass();
+        sqlDao = getDBI().onDemand(PersistentBusSqlDao.class);
+    }
+
+    @BeforeMethod(groups = "slow")
+    public void beforeMethod() throws Exception {
+        super.beforeMethod();
+        final List<BusEventModelDao> ready = sqlDao.getReadyEntries(clock.getUTCNow().toDate(), 100, null, "bus_events");
+        assertEquals(ready.size(), 0);
+    }
+
+    @Test(groups = "slow")
+    public void testReapEntries() {
+        final PersistentBusConfig config = createConfig();
+        queue = new DBBackedQueueWithPolling<BusEventModelDao>(clock, dbi, PersistentBusSqlDao.class, config, "testReapEntries", metricRegistry);
+        final DateTime now = clock.getUTCNow();
+
+        // Insert ready entry (not yet reapable)
+        sqlDao.insertEntry(createEntry(1L, CreatorName.get(), null, now, PersistentQueueEntryLifecycleState.AVAILABLE), config.getTableName());
+        // Insert ready entry from another node (not yet reapable)
+        sqlDao.insertEntry(createEntry(2L, "thatOtherNode", null, now, PersistentQueueEntryLifecycleState.AVAILABLE), config.getTableName());
+        // Insert in-processing entry (not yet reapable)
+        sqlDao.insertEntry(createEntry(3L, CreatorName.get(), CreatorName.get(), now, PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
+        // Insert in-processing entry from another node (not yet reapable)
+        sqlDao.insertEntry(createEntry(4L, "thatOtherNode", "thatOtherNode", now, PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
+        // Insert in-processing entry (reapable)
+        sqlDao.insertEntry(createEntryForReaping(5L, CreatorName.get(), CreatorName.get(), now, config.getReapThreshold().getMillis(), PersistentQueueEntryLifecycleState.IN_PROCESSING), config.getTableName());
```

Comment:
`(reapable)` -> Not reapable by this node though.

Reply:
Good catch, see ead0a8f7b421691999d6489b356ccb9ff43d8db4.

It's actually a left over from d3423bd1af270bfaff52cc69e7ad26c2bef3894c: the entry was legacy-repeable before the change.